proto/catalog.proto

```diff
@@ -62,9 +62,16 @@ message StreamSourceInfo {
   SchemaRegistryNameStrategy name_strategy = 10;
   optional string key_message_name = 11;
   plan_common.ExternalTableDesc external_table = 12;
-  // Whether the stream source is a cdc source streaming job.
-  // We need this field to differentiate the cdc source job until we fully implement risingwavelabs/rfcs#72.
-  bool cdc_source_job = 13;
+  // Whether the stream source has a streaming job.
+  // This is related with [RFC: Reusable Source Executor](https://github.com/risingwavelabs/rfcs/pull/72).
+  // Currently, the following sources have streaming jobs:
+  // - Direct CDC sources (mysql & postgresql)
+  // - MQ sources (Kafka, Pulsar, Kinesis, etc.)
+  bool has_streaming_job = 13;
+  // Only used when `has_streaming_job` is `true`.
+  // If `false`, `requires_singleton` will be set in the stream plan.
+  bool is_distributed = 15;
+  reserved "cdc_source_job"; // deprecated
```

Comment:
Deprecating `cdc_source_job` and adding `has_streaming_job` will break backward compability of existing CDC jobs, right?

Why not renaming `cdc_source_job` to `has_streaming_job`?

Reply:
> Why not renaming cdc_source_job to has_streaming_job?

This is indeed the change I made. Number `13` is renamed. 

> Changing the field name could be breaking under SQL meta backend.

Indeed. 😕 Need to come up with another plan.

Reply:
do you already have one? Keeping it or introducing another one dedicated to this case both LGTM

Reply:
New plan:
- keep `cdc_source_job`. New & old CDC sources will both set this field. This field is only used to set `require_singleton`
- Change `is_shared` to `optional` (explicit presence). Old CDC sources will have this field set to `None`. New CDC sources will set to `Some(true)`.
  - added method `is_shared_compatibile` to test `cdc_source_job` first. TBH, this is a little awkward. Better ideas are welcome.
  - Strictly speaking, it's not necessary to use `optional`, `(cdc_source_job, is_shared) = (true,false)` can also work. But I feel this can make it (just slightly) less error-prone.

---
Old plan:
- Rename `cdc_source_job` to `is_shared`. This is more elegant, but breaks JSON compatibility for SQL meta backend.
- Added `is_distributed` to set `require_singleton` in fragmenter for CDC source.

Reply:
> Rename `cdc_source_job` to `is_shared`. This is more elegant, but breaks JSON compatibility for SQL meta backend.


Maybe off-topic... I think storing protobuf into JSON sounds bad.

<img width="1010" alt="image" src="https://github.com/risingwavelabs/risingwave/assets/10192522/336121d2-c040-4493-9ad0-dbfdd5abba48">

https://protobuf.dev/programming-guides/dos-donts/#text-format-interchange